pkg/collect/host_collector.go

```diff
@@ -81,3 +110,130 @@ func hostCollectorTitleOrDefault(meta troubleshootv1beta2.HostCollectorMeta, def
+	defer cancel()
+
+	// Get all the nodes where we should run.
+	nodes, err := listNodesNamesInSelector(ctx, client, params.LabelSelector)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get the list of nodes matching a nodeSelector")
+	}
+
+	if params.NamePrefix == "" {
+		params.NamePrefix = remoteCollectorNamePrefix
+	}
+
+	result, err := runRemote(ctx, runner, nodes, params.HostCollector, names.SimpleNameGenerator, params.NamePrefix, params.Namespace)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to run collector remotely")
+	}
+
+	for k, v := range result {
+		if curBytes, ok := allCollectedData[k]; ok {
+			var curResults map[string]string
+			if err := json.Unmarshal(curBytes, &curResults); err != nil {
+				params.ProgressChan <- errors.Errorf("failed to read existing results for collector %s: %v\n", params.Title, err)
+				continue
+			}
+			var newResults map[string]string
+			if err := json.Unmarshal(v, &newResults); err != nil {
+				params.ProgressChan <- errors.Errorf("failed to read new results for collector %s: %v\n", params.Title, err)
+				continue
+			}
+			for file, data := range newResults {
+				curResults[file] = data
+			}
+			combinedResults, err := json.Marshal(curResults)
+			if err != nil {
+				params.ProgressChan <- errors.Errorf("failed to combine results for collector %s: %v\n", params.Title, err)
+				continue
+			}
+			allCollectedData[k] = combinedResults
+		} else {
+			allCollectedData[k] = v
+		}
+
+	}
+
+	output := NewResult()
+
+	// save the first result we find in the node and save it
+	for node, result := range allCollectedData {
+		var nodeResult map[string]string
+		if err := json.Unmarshal(result, &nodeResult); err != nil {
+			return nil, errors.Wrap(err, "failed to marshal node results")
+		}
+
+		for file, collectorResult := range nodeResult {
+			directory := filepath.Dir(file)
+			fileName := filepath.Base(file)
+			// expected file name for remote collectors will be the normal path separated by / and the node name
+			output.SaveResult(params.BundlePath, fmt.Sprintf("%s/%s/%s", directory, node, fileName), bytes.NewBufferString(collectorResult))
+		}
+	}
+
+	// check if NODE_LIST_FILE exists
+	_, err = os.Stat(constants.NODE_LIST_FILE)
+	// if it not exists, save the nodes list
+	if err != nil {
+		nodesBytes, err := json.MarshalIndent(HostOSInfoNodes{Nodes: nodes}, "", " ")
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to marshal host os info nodes")
+		}
+		output.SaveResult(params.BundlePath, constants.NODE_LIST_FILE, bytes.NewBuffer(nodesBytes))
+	}
+	return output, nil
+}
+
+func runRemote(ctx context.Context, runner runner, nodes []string, collector *troubleshootv1beta2.HostCollect, nameGenerator names.NameGenerator, namePrefix string, namespace string) (map[string][]byte, error) {
+	g, ctx := errgroup.WithContext(ctx)
+	results := make(chan map[string][]byte, len(nodes))
+
+	for _, node := range nodes {
```

Comment:
I am thinking limit the collector to run 1 node per time to reduce the kube-api limit. Not 100% sure, it is worthing
```
func runRemote(ctx context.Context, runner runner, nodes []string, collector *troubleshootv1beta2.HostCollect, nameGenerator names.NameGenerator, namePrefix string, namespace string) (map[string][]byte, error) {
	g, ctx := errgroup.WithContext(ctx)
	results := make(chan map[string][]byte, len(nodes))
	wg := &sync.WaitGroup{}
	workerLimit := make(chan struct{}, 1)

	for _, node := range nodes {
		node := node
		workerLimit <- struct{}{} // limit the number of concurrent workers
		wg.Add(1)

		g.Go(func() error {
			// May need to evaluate error and log warning.  Otherwise any error
			// here will cancel the context of other goroutines and no results
			// will be returned.
			defer wg.Done()
			defer func() { <-workerLimit }()
			return runner.run(ctx, collector, namespace, nameGenerator.GenerateName(namePrefix+"-"), node, results)
		})
	}

	// Wait for all collectors to complete or return the first error.
	if err := g.Wait(); err != nil {
		return nil, errors.Wrap(err, "failed remote collection")
	}
	close(results)

	wg.Wait()

	output := make(map[string][]byte)
	for result := range results {
		r := result
		for k, v := range r {
			output[k] = v
		}
	}

	return output, nil
}

```

Reply:
I wasn't using the same k8s REST config that the standard in cluster collection used. Now that I am, I'm not seeing the throttling anymore. It's possible we would still need to keep an eye on this in the future for very large clusters though